message_filters uses Python `@dataclass`, which is not available in Humble/RHEL 8/Python 3.6

### Generated by Generative AI

_No response_

### Operating System:

RHEL 8

### ROS version or commit hash:

Humble

### RMW implementation (if applicable):

_No response_

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

<details><summary>ros2 doctor --report</summary>

```console
<COPY OUTPUT HERE>
```
</details>


### Steps to reproduce issue

Run tests for `message_filters` @ Humble on RHEL 8.

### Expected behavior

Tests pass.

### Actual behavior

Tests fail with:

```
../../src/ros2/message_filters/test/test_message_filters_chain.py:32: in <module>
    from message_filters import Cache, Chain, Subscriber
../../install/message_filters/lib/python3.6/site-packages/message_filters/__init__.py:36: in <module>
    from dataclasses import dataclass
E   ModuleNotFoundError: No module named 'dataclasses'
```

### Additional information

[Humble targets RHEL 8, which has Python 3.6](https://reps.openrobotics.org/rep-2000/#humble-hawksbill-may-2022---may-2027). However, `dataclasses` was only added in Python 3.7: https://docs.python.org/3/library/dataclasses.html. That import was added in `message_filters` in https://github.com/ros2/message_filters/pull/216, which is a backport of https://github.com/ros2/message_filters/pull/213.

We should find an alternative to `dataclasses`/`@dataclass` for Humble.

--- a/src/message_filters/__init__.py
+++ b/src/message_filters/__init__.py
@@ -32,8 +32,6 @@ Message Filter Objects
 
 """Message Filter Objects."""
 
-from bisect import insort_right
-from dataclasses import dataclass
 from functools import reduce
 import itertools
 import threading
@@ -238,10 +236,14 @@ class Chain(SimpleFilter):
     to the callback you've registered with Chain::registerCallback.
     """
 
-    @dataclass
     class FilterInfo:
-        message_filter: any
-        connection_callback_index: int
+        def __init__(
+            self,
+            message_filter: any,
+            connection_callback_index: int,
+        ):
+            self.message_filter = message_filter
+            self.connection_callback_index = connection_callback_index
 
     def __init__(self, message_filter=None):
         SimpleFilter.__init__(self)

